OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -408,29 +408,35 @@ public void Load(Actor self, Actor a)
 		}
 
 		void INotifyKilled.Killed(Actor self, AttackInfo e)
-		{
-			if (Info.EjectOnDeath)
-				while (!IsEmpty() && CanUnload(BlockedByActor.All))
+ 		{
+			if (Info.EjectOnDeath && self.IsAtGroundLevel() && (!checkTerrainType || Info.UnloadTerrainTypes.Contains(self.World.Map.GetTerrainInfo(self.Location).Type)))
```

Comment:
This should check if the map contains the cell before trying to get the terrain info.

Reply:
Unload doesn't contain such checks 🤔

Reply:
Then it probably should. It is unlikely that something unloads outside of the map, but that would currently crash, wouldn't it?

Reply:
check added